account_payment_order/models/account_payment_order.py

```diff
@@ -514,3 +514,20 @@ def generate_move(self):
         trfmoves = self._prepare_trf_moves()
         for hashcode, blines in trfmoves.items():
             self._create_reconcile_move(hashcode, blines)
+
+    @api.multi
+    def _all_lines_reconciled(self):
+        """
+        Return true if all payment lines' counterpart lines are reconciled too
+        Also return true if the counterpart account cannot be reconciled
+        """
+        result = True
+        for this in self:
+            move_lines = self.env['account.move.line'].search([(
+                'bank_payment_line_id', 'in', this.mapped('bank_line_ids').ids
+            )]).mapped('move_id.line_ids')
+            result &= all(
```

Comment:
I know this is currently only called on orders with move lines, but in case this gets a life of its own it might be useful to check ifthere are any move lines to begin with.